### PR TITLE
Components: Add stories for Icon and IconButton

### PR DESCRIPTION
**Pull request by ItsJonQ**
Screen Shot 2019-10-09 at 11 31 03 AM

This update adds some initial Storybook stories for the Icon and IconButton
component.

Most of the stories were derived from the component's README.md documentation.
Additional stories were added to showcase prop variation (e.g. `Icon` size),
as well as composition (e.g. `IconButton` grouping).

In order to render certain UI for the stories, some simple stateless
functional components were created in the `.stories.js` file.

### CODE REVIEW

**Comment by youknowriad** [comment]
@epiqueras do we have a live deploy in netlify for these PRs? How can we get the URL?

**Comment by ItsJonQ** [comment]
@gziolo Thanks for that context! In that case, should I remove the `IconButton` stories for now?

**Comment by gziolo** [comment]
> @gziolo Thanks for that context! In that case, should I remove the `IconButton` stories for now?

I guess it's fine to proceed. We will have to keep `IconButton` in WordPress core for very long even when we decide to merge it into `Button` :)

**Comment by ItsJonQ** [comment]
> We will have to keep IconButton in WordPress core for very long even when we decide to merge it into Button :)

😅

**Comment by epiqueras** [comment]
@youknowriad The link is in the PR checks, `https://deploy-preview-${PR_NUMBER}--gutenberg-playground.netlify.com`.

You'll need to manually add `/design-system/components` at the end, because there's no link to it yet from the playground.

I just updated the Netlify config to match the new order of build scripts though, so we'll need a new commit in this PR to trigger a new build.

**Comment by youknowriad** [comment]
@epiqueras https://deploy-preview-17868--gutenberg-playground.netlify.com/design-system/components This doesn't seem to work?

**Comment by epiqueras** [comment]
@youknowriad 

>I just updated the Netlify config to match the new order of build scripts though, so we'll need a new commit in this PR to trigger a new build.

File: packages/components/src/icon/stories/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+const IconSizeLabel = ( { size } ) => <div style={ { fontSize: 12 } }>{ size }px</div>;
+
+export const _default = () => (
+	<div>
+		<Icon icon="screenoptions" />
+		<IconSizeLabel size={ 24 } />
+	</div>
+);
+
+export const Sizes = () => {
```
Comment (epiqueras) [inline packages/components/src/icon/stories/index.js]: I think the convention is to use `camelCase` for story names: https://storybook.js.org/docs/formats/component-story-format/.

I don't have a strong opinion, but we used that in the stories we already merged so it's just for consistency.

File: packages/components/src/icon-button/stories/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import IconButton from '../';
+
+export default { title: 'IconButton', component: IconButton };
+
+export const _default = () => <IconButton icon="ellipsis" label="More" />;
+
+export const Grouped = () => {
+	const GroupContainer = ( props ) => (
+		<div style={ { display: 'inline-flex' } } { ...props } />
+	);
```
Comment (epiqueras) [inline packages/components/src/icon-button/stories/index.js]: We could remove the props forwarding here to make things simpler.

**Review by epiqueras — COMMENTED** [review]
Thanks for moving this forward! :smile: :rocket:

File: packages/components/src/icon/stories/index.js (hunk 1 above)
Comment (ItsJonQ) [reply packages/components/src/icon/stories/index.js]: @epiqueras Thank you!

File: packages/components/src/icon-button/stories/index.js (hunk 2 above)
Comment (ItsJonQ) [reply packages/components/src/icon-button/stories/index.js]: Yes! Can do. Thank you! <3

**Review by epiqueras — APPROVED** [review]

**Pull request merged**